From 93be42e3d9c109ed338c936dfb3319e18ae17ed8 Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 15:21:55 -0400
Subject: [infiniband] ocrdma: Fix warnings about pointer <-> integer casts

Message-id: <92b715681f82f3d4b9258340be1ea287b9e05e50.1411571481.git.dledford@redhat.com>
Patchwork-id: 96650
O-Subject: [Patch RHEL7 20/48] RDMA/ocrdma: Fix warnings about pointer <-> integer casts
Bugzilla: 1081067 1088577
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Roland Dreier <roland@purestorage.com>

We should cast pointers to and from unsigned long to turn them into ints.

Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 7a1e89d8b793feda96754fd3ba5079c4ba3862ee)
Signed-off-by: Doug Ledford <dledford@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c b/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
index 7370b79..edf6211 100644
--- a/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
+++ b/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
@@ -2025,8 +2025,8 @@ static int ocrdma_build_fr(struct ocrdma_qp *qp, struct ocrdma_hdr_wqe *hdr,
  fast_reg->num_sges = wr->wr.fast_reg.page_list_len;
  fast_reg->size_sge =
   get_encoded_page_size(1 << wr->wr.fast_reg.page_shift);
- mr = (struct ocrdma_mr *)qp->dev->stag_arr[(hdr->lkey >> 8) &
-  (OCRDMA_MAX_STAG - 1)];
+ mr = (struct ocrdma_mr *) (unsigned long)
+  qp->dev->stag_arr[(hdr->lkey >> 8) & (OCRDMA_MAX_STAG - 1)];
  build_frmr_pbes(wr, mr->hwmr.pbl_table, &mr->hwmr);
  return 0;
 }
@@ -2860,7 +2860,8 @@ struct ib_mr *ocrdma_alloc_frmr(struct ib_pd *ibpd, int max_page_list_len)
   goto mbx_err;
  mr->ibmr.rkey = mr->hwmr.lkey;
  mr->ibmr.lkey = mr->hwmr.lkey;
- dev->stag_arr[(mr->hwmr.lkey >> 8) & (OCRDMA_MAX_STAG - 1)] = (u64)mr;
+ dev->stag_arr[(mr->hwmr.lkey >> 8) & (OCRDMA_MAX_STAG - 1)] =
+  (unsigned long) mr;
  return &mr->ibmr;
 mbx_err:
  ocrdma_free_mr_pbl_tbl(dev, &mr->hwmr);
-- 
1.7.1